### PR TITLE
Fix many issues reported by clang-tidy

### PR DESCRIPTION
**Pull request by nerzhul**
We have many issues in code related to some performance to float <-> double.
Clang-tidy reported it in performance-type-promotion-in-math-fn

I fixed many of them. It's not ready for a promote to blocking

Also fix some value which should be const-ref

### CODE REVIEW

File: src/mapgen/mapgen_carpathian.cpp [hunk 1]
```diff
@@ -342,8 +342,8 @@ float MapgenCarpathian::terrainLevelAtPoint(s16 x, s16 z)
 			std::fmax(std::fmin(hill1, hill2), std::fmin(hill3, hill4));
 
 		// Rolling hills
-		float hill_mnt = hilliness * pow(n_hills, 2.f);
-		float hills = pow(hter, 3.f) * hill_mnt;
+		float hill_mnt = hilliness * std::pow(n_hills, 2.f);
+		float hills = std::pow(hter, 3.f) * hill_mnt;
```
Comment (paramat) [inline src/mapgen/mapgen_carpathian.cpp]: You've missed some 'pows' in following lines 350 354

**Comment by paramat** [comment]
Mapgen stuff seems ok.

**Comment by nerzhul** [comment]
Oh i fixed many point not all :) Will do more points in another pr

**Comment by paramat** [comment]
Ok.

File: src/mapgen/mg_ore.cpp [hunk 2]
```diff
@@ -345,7 +345,8 @@ void OreBlob::generate(MMVManip *vm, int mapseed, u32 blockseed,
 			float ydist = (s32)y1 - (s32)csize / 2;
 			float zdist = (s32)z1 - (s32)csize / 2;
 
-			noiseval -= (sqrt(xdist * xdist + ydist * ydist + zdist * zdist) / csize);
+			noiseval -= (std::sqrt(xdist * xdist + ydist * ydist + zdist * zdist) /
```
Comment (sfan5) [inline src/mapgen/mg_ore.cpp]: the outer parentheses are useless here

**Pull request merged**